m4/curl-functions.m4

```diff
@@ -3394,6 +3394,60 @@ AC_DEFUN([CURL_CHECK_FUNC_MEMRCHR], [
 ])
 
 
+dnl CURL_CHECK_FUNC_SETSOCKOPT_SOL_IP
+dnl -------------------------------------------------
+dnl Verify if setsockopt with the SOL_IP command is
+dnl available, can be compiled, and seems to work. If
+dnl all of these are true, then HAVE_SETSOCKOPT_SOL_IP
+dnl will be defined.
+
+AC_DEFUN([CURL_CHECK_FUNC_SETSOCKOPT_SOL_IP], [
+  #
+  tst_compi_setsockopt_sol_ip="unknown"
+  tst_allow_setsockopt_sol_ip="unknown"
+  #
+  AC_MSG_CHECKING([if setsockopt SOL_IP is compilable])
+  AC_COMPILE_IFELSE([
+    AC_LANG_PROGRAM([[
+      $curl_includes_sys_socket
+      #include <netinet/ip.h>
+    ]],[[
+      if(0 != setsockopt(0, SOL_IP, IP_TOS, 0, 0))
```

Comment:
Why check for `SOL_IP` support for setsockopt when the code is using `IPPROTO_IP` in that argument?

Reply:
According to some [posts](https://lists.gnu.org/archive/html/autoconf/2010-03/msg00119.html) I found, maybe the other way around is more portable?

Reply:
Done.